swagger.yaml

```diff
@@ -1,42 +1,292 @@
+            Invalid request, reasons can be one of the following:
+              * image id was incorrect
+              * collection id was incorrect
+              * malformed body
+              * provided image had an invalid parameter
+        401:
+          description: "Unauthorised to update image metadata"
+        404:
+          $ref: '#/responses/NotFound'
+        409:
+          description: "Cannot update an image for the provided collection ID because it is already published"
         500:
           $ref: '#/responses/InternalError'
 
 responses:
+
   InternalError:
     description: "Failed to process the request due to an internal error"
+    
+  NotFound:
+    description: "Requested item cannot be found"
+    
+  InvalidRequestError:
+    description: "Failed to process the request due to invalid request"
 
 definitions:
-  HelloResponse:
+
+  Images:
+    description: "A list of images"
     type: object
     properties:
-      message:
+      count:
+        description: "The number of images returned"
+        readOnly: true
+        type: integer
+        example: 1
+      items:
+        type: array
+        items:
+          $ref: '#/definitions/Image'
+      limit:
+        description: "The number of images requested"
+        type: integer
+      offset:
+        description: "The first row of images to retrieve, starting at 0. Use this parameter as a pagination mechanism along with the limit parameter"
+        type: integer
+      total_count:
+        description: "The total number of images"
+        readOnly: true
+        type: integer
+        example: 1
+
+  Image:
+    type: object
+    description: "Image metadata model"
+    properties:
+      id:
+        type: string
+        description: "Image metadata unique identifier"
+        example: "042e216a-7822-4fa0-a3d6-e3f5248ffc35"
+      collection_id:
+        type: string
+        description: "Collection unique identifier corresponding to this image"
+        example: "5557dcd9-bf58-4a67-94f7-2343569834cc"
+      state:
+        type: string
+        description: |
+          The state of the image, can only be one of the following:
+          * created
+          * published
+        example: "published"
+      filename:
         type: string
-        description: "Message returned by hello world endpoint"
-        example: "Hello, world!"
\ No newline at end of file
+        description: "image's file name"
+        example: "imageName.jpg"
+      license:
+        type: object
+        description: "license that defines the third party permissions for this image"
+        properties:
+          title:
+            type: string
+            description: "Title of the license"
+            example: "Open Government Licence v3.0"
+          href: 
+            type: string
+            description: "Link to the license content"
+            example: "https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
+      original:
```

Comment:
I think this term is from the original story, but seeing it in the spec I wonder if `upload` would be better as it would align to the `downloads`? Not precious on this either way, this is just mostly thinking out loud.